src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -81,6 +79,7 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         } else {
             this.exerciseService.findResultsForExercise(this.exerciseId).subscribe((exerciseResponse: HttpResponse<Exercise>) => {
                 this.exercise = exerciseResponse.body;
+                this.sortParticipationsFinishedFirst(this.exercise.participations);
```

Comment:
I think it should be like this?
this.exercise.participations = this.filterParticipations(cachedParticipations);

Reply:
In the else block here `cachedParticipations` is empty and the participations come with the exercise in the response from the server. So we should use `this.exercise.participations` instead of `cachedParticipations`. But we could additionally filter these participations, you are right. It should not be necessary as the server makes sure that the user only gets his participations and no uncompleted results. To be on the safe side and for consistency reasons I changed the line to `this.exercise.participations = this.filterParticipations(this.exercise.participations);`. Thanks :)